spot_wrapper/calibration/calibration_util.py

```diff
@@ -0,0 +1,916 @@
+
+            The relationship between `desired_stereo_pairs` and `images` is crucial. Each tuple in
+            `desired_stereo_pairs` should be in the form
+            `(origin_camera_idx, reference_camera_idx)`,
+            where `origin_camera_idx` and `reference_camera_idx` correspond to the
+            indices of the cameras in the `images` array. For each stereo pair,
+            the function attempts to calibrate the `origin_camera_idx`
+            relative to the `reference_camera_idx`. If no pairs are provided,
+            the function defaults to registering
+            all cameras to the first camera (index `0`).
+
+        charuco_board (cv2.aruco_CharucoBoard, optional): The Charuco board configuration used
+            for calibration. Defaults to SPOT_DEFAULT_CHARUCO.
+
+        aruco_dict (cv2.aruco_Dictionary, optional): The Aruco dictionary used
+            to detect the markers. Defaults to SPOT_DEFAULT_ARUCO_DICT.
+
+        camera_matrices (Optional[Dict], optional): A dictionary mapping camera indices
+            to their existing camera matrices. If a matrix is not provided for a camera,
+            it will be computed during calibration
+            if that camera is part of a stereo pair. Defaults to an empty dictionary.
+
+        dist_coeffs (Optional[Dict], optional): A dictionary mapping camera indices
+            to their distortion coefficients.
+            If coefficients are not provided for a camera, they will be computed during calibration.
+            Defaults to an empty dictionary.
+
+    Raises:
+        ValueError: Raised if fewer than two cameras are provided, as stereo calibration
+            requires at least two cameras.
+        ValueError: Raised if a stereo pair includes the same camera twice.
+        ValueError: Raised if a camera in a stereo pair cannot be calibrated.
+
+    Returns:
+        Dict: A dictionary containing the calibrated intrinsic and extrinsic parameters
+            for the specified stereo pairs.
+        The keys are formatted as "origin_camera_idx_reference_camera_idx",
+            and the values contain the calibration data.
+
+    """
+    calibration = {}
+    num_cams = images.shape[1]
+    if num_cams == 0 or num_cams == 1:
+        raise ValueError("Stereo Madness requires more than one camera. Use the monocular method instead.")
+    else:  # More than one camera - Multi-Stereo Madness ;p
+        for idx in range(num_cams):
+            try:
+                _ = camera_matrices[idx]
+            except KeyError:
+                camera_matrices[idx] = None
+            try:
+                _ = dist_coeffs[idx]
+            except KeyError:
+                dist_coeffs[idx] = None
+
+        if desired_stereo_pairs is None:
+            desired_stereo_pairs = []
+            for idx in range(1, num_cams):
+                desired_stereo_pairs.append((0, idx))  # just register all to 0th cam
```

Comment:
In the string you mention the tuple being (origin, reference) and if I understand the string right you say that means you'll get extrinsics that will calibrate origin relative to reference (i.e. reference is the base frame?). You also mention that by default everything will be relative to the 0th camera, but here you are defaulting 0 relative to all other cameras. I just want to make sure this is the intent. Its possible I misread your comment and previous examples.

Reply:
Sorry if that wasn't clear

The origin camera is the base frame, and the reference frame is with reference to the origin camera (base frame).

I think a clearer comment might be 

```desired_stereo_pairs.append((0, idx)) # register everything IN/FROM 0th coordinate system``` 

or alternatively 

```desired_stereo_pairs.append((idx, 0)) # just register all to 0th cam```

Let me know what you think makes more sense for a default as I agree that it's a little unclear.

Maybe I can also just add a comment about origin vs. reference

Reply:
Sounds good! yeah I think I was thrown off because in the default for the --stereo_pairs you set (1,0) as the default. Line: 78 in the calibrate_multistereo_camera_cli.py but yeah I think the comment is correct there.

Reply:
Updated this to be 

```
if desired_stereo_pairs is None:
            desired_stereo_pairs = []
            for idx in range(1, num_cams):
                """
                Desired stereo pairs is a mapping, where the 0th index is the origin (base)
                frame, where the reference frame is defined within the origin (base) frame's
                coordinate system. The reference frame is the 1st index.

                By Default, if desired_stereo_pairs is None,
                then each camera's intrinsic is defined in the 0th camera's frame,
                so this is set as the origin for all stereo calibrations
                """
                desired_stereo_pairs.append((0, idx))  # just register all in 0th cam frame
```


Yeah default is set to (1,0) bc the rgb feed here is 0th index in capture_images, and depth is registered to rgb. Idk why but I felt like rgb being index 0 made more sense over depth (sorry depth I still love you)